### ISSUE TITLE
[Bug]: Invalid/GEN9 mods are still applied when using max_vehicle / spawn_maxed

### ISSUE THREAD

**Issue by NjuuOwO** [Yimura/YimMenu#2001]
### Describe the bug

IDK if it's left on purpose, but it still is here.

https://github.com/YimMenu/YimMenu/assets/45913099/6bc8ce10-8d72-459f-ac90-447de7c92485

### Steps To Reproduce

1. Spawn maxed vehicle / Max Vehicle

### Expected Behavior

don't apply gen9 mod

### Logs and/or screenshots

```
[19:07:43.2501135][INFO/main.cpp:65] Yim's Menu Initializing
[19:07:43.2501422][INFO/main.cpp:66] Git Info
	Branch:	master
	Hash:	81e43b51d7049a40b49d826d09bdc3828b1c7bb3
	Date:	Sat Aug 19 13:02:26 2023
[19:07:43.2501567][DEBUG/thread_pool.cpp:21] Resizing thread pool from 0 to 4
[19:07:43.2863655][INFO/main.cpp:69] Thread pool initialized.
[19:07:43.2894077][INFO/main.cpp:72] Settings Loaded.
[19:07:43.2894162][DEBUG/thread_pool.cpp:95] Thread 17932 executing settings.cpp:18
[19:07:43.2899169][INFO/pointers.hpp:94] GTA5 pointers cache is up to date, using it.
[19:07:43.2899277][INFO/pointers.cpp:1687] Loading pointers instance from cache
[19:07:43.2899290][INFO/pointers.cpp:1696] Pointers cache: Loading 200 fields from the cache
[19:07:43.2899372][DEBUG/module.cpp:50] Waiting for socialclub.dll...
[19:07:43.2901581][INFO/pointers.hpp:94] Social Club pointers cache is up to date, using it.
[19:07:43.2901615][INFO/pointers.cpp:1687] Loading pointers instance from cache
[19:07:43.2901631][INFO/pointers.cpp:1696] Pointers cache: Loading 5 fields from the cache
[19:07:43.2902010][INFO/main.cpp:75] Pointers initialized.
[19:07:43.2902805][INFO/main.cpp:78] Byte Patch Manager initialized.
[19:07:43.6406064][INFO/main.cpp:81] Renderer initialized.
[19:07:43.7895757][INFO/main.cpp:85] Fiber pool initialized.
[19:07:44.0714382][INFO/main.cpp:88] Translation Service initialized.
[19:07:44.0822783][INFO/main.cpp:91] Hooking initialized.
[19:07:44.0879987][DEBUG/gta_data_service.cpp:162] Loading data from cache.
[19:07:44.0880101][INFO/gta_data_service.cpp:174] Loading 1029 peds from cache.
[19:07:44.0882979][INFO/gta_data_service.cpp:192] Loading 815 vehicles from cache.
[19:07:44.0885165][INFO/gta_data_service.cpp:209] Loading 117 weapons from cache.
[19:07:44.0885184][INFO/gta_data_service.cpp:210] Loading 378 weapon components from cache.
[19:07:44.0885336][DEBUG/gta_data_service.cpp:168] Loaded all data from cache.
[19:07:44.0922930][INFO/tunables_service.cpp:21] Loading tunables from cache
[19:07:44.0923146][INFO/main.cpp:111] Registered service instances...
[19:07:44.0923180][DEBUG/thread_pool.cpp:95] Thread 16804 executing tunables_service.cpp:24
[19:07:44.0926984][INFO/main.cpp:132] Scripts registered.
[19:07:44.1150432][INFO/main.cpp:135] Hooking enabled.
[19:07:44.1159285][INFO/main.cpp:138] Dynamic native hooker initialized.
[19:07:44.1528644][INFO/lua_module.cpp:226] Loaded Alestarov_Menu_V2.1b_1.67.lua
[19:07:44.1579342][INFO/lua_module.cpp:226] Loaded CayoSpeedrunPrep.lua
[19:07:44.1623334][INFO/lua_module.cpp:226] Loaded shift-drift.lua
[19:07:44.1678190][INFO/lua_module.cpp:226] Loaded Ultimate-Menu.lua
[19:07:44.1723528][INFO/lua_module.cpp:226] Loaded YimNightClub_[unknowncheats.me]_.lua
[19:07:44.1723820][INFO/main.cpp:141] Lua manager initialized.
[19:07:44.1723996][DEBUG/thread_pool.cpp:95] Thread 16804 executing player_database_service.cpp:276
[19:07:44.3140749][INFO/backend.cpp:51] Starting script: Self
[19:07:44.3140880][INFO/backend.cpp:76] Starting script: Weapons
[19:07:44.3141024][INFO/backend.cpp:99] Starting script: Vehicles
[19:07:44.3141152][INFO/backend.cpp:115] Starting script: Miscellaneous
[19:07:44.3141229][INFO/backend.cpp:135] Starting script: Remote
[19:07:44.3141299][INFO/backend.cpp:150] Starting script: Ls customs
[19:07:44.3141316][INFO/backend.cpp:162] Starting script: Rainbow paint
[19:07:44.3141339][INFO/backend.cpp:174] Starting script: Disable Control Action
[19:07:44.3141377][INFO/backend.cpp:187] Starting script: World
```

### Store

Rockstar

### Additional context

In my understanding this line of code (vehicle.hpp:597) doesn't work properly?

```
if (VEHICLE::IS_VEHICLE_MOD_GEN9_EXCLUSIVE(veh, slot, mod))
{
	continue;
}
```

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by holysnipz** [comment]
looks like it didn't check for HSW Turbo (modType/slot = 34)?

![image](https://github.com/YimMenu/YimMenu/assets/9103583/46e98268-0d15-42ad-9db9-3ff126a54d1f)

----------------------------
EDIT
and apparently the same slot is using as Shifter Leavers on Benny's cars as well.
*(nobody cares about this get maxed I know)*

![image](https://github.com/YimMenu/YimMenu/assets/9103583/947a58e0-7817-4e6c-ac3c-ed9ddd25b8e8)

so, skipping slot 34 should be fine I guess. if someone wants, they can just use LSC for it

**Comment by ScottiePimp** [comment]
Just add a "if (abc == 34) continue" then.

**Comment by NjuuOwO** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/45913099/22662ddd-43ac-4f9b-ad27-1ab54838eac2)

Not sure if every invalid mod is at 34 slot for every vehicle

@holysnipz Where'd you get that script for reading mods?

**Comment by FIying-Scotsman** [comment]
> ![image](https://user-images.githubusercontent.com/45913099/262330248-22662ddd-43ac-4f9b-ad27-1ab54838eac2.png)
> 
> Not sure if every invalid mod is at 34 slot for every vehicle
> 
> @holysnipz Where'd you get that script for reading mods?

From memory there's also an invalid spoiler and front splitter for one those Koenigsegg (Entity MT?) and Porsche looking cars  and I think Issi8. Can't remember the mod slot however.

**Comment by NjuuOwO** [comment]
So it's basically all vehicles that have GEN9 mods

**Comment by ScottiePimp** [comment]
> > ![image](https://user-images.githubusercontent.com/45913099/262330248-22662ddd-43ac-4f9b-ad27-1ab54838eac2.png)
> 
> > 
> 
> > Not sure if every invalid mod is at 34 slot for every vehicle
> 
> > 
> 
> > @holysnipz Where'd you get that script for reading mods?
> 
> 
> 
> From memory there's also an invalid spoiler and front splitter for one those Koenigsegg (Entity MT?) and Porsche looking cars  and I think Issi8. Can't remember the mod slot however.

You only need to worry about the HSW vehicles. I have a check for these in my menu and other vehicles, such as the Entity you mentioned, don't cause any exceptions/crashes. You can very easily debug which component the HSW is at (if its not 34 on all, I'm not at PC so I can't check how mine looks like).

**Comment by NjuuOwO** [comment]
I've done some debugging and seems like it detects GEN9 mods but doesn't actually exclude from applying them.

EDIT: modType is off by 2? 34 is invalid Turbo HSW mod for hakuchou2, but max_vehicle returns this for me:

>[INFO/vehicle.hpp:599] Slot 36 excluded by IS_VEHICLE_MOD_GEN9_EXCLUSIVE.

EDIT2: Sorry, it's not off, and it DOES exclude them, i'm just dumb. Slot 34 (HSW Turbo) on hakuchou2 just isn't recognized as invalid mod, thus it is applied.
BUT for e.g. Entity MT, slot 34 (HSW Turbo) isn't applied, but 2 other invalids are set.

So turns out HSW vehicles have some invalid parts not included in GEN9_EXCLUSIVE_ASSETS_VEHICLES_FILE

**Comment by NjuuOwO** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/45913099/1a1dcf07-1f84-467d-b3b5-0d8e1dbe7518)
I've made a list with invalid mods, that are applied by `vehicle::max_vehicle()`, and best for the slot. no more vehicles have invalid mods when maxed. i hope it's readable
First `modindex` value is applied by default by ^^^.
modtype 34 : 2,1,0 don't cause crash until 3 is first applied

Maybe creating some sort of ban list will do the job? I will let the people speak

**Comment by Yimura** [comment]
Fixed in #2063
